src/Infrastructure/LeanCode.AI/CQRSAIFunction.cs

```diff
@@ -0,0 +1,209 @@
+using System.Diagnostics;
+using System.Reflection;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Text.Json.Schema;
+using LeanCode.AI.Contracts;
+using LeanCode.Contracts;
+using LeanCode.CQRS.RemoteHttp.Client;
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.DependencyInjection;
+using static LeanCode.AI.Contracts.McpToolHints;
+using IOException = System.InvalidOperationException;
+
+namespace LeanCode.AI;
+
+public abstract class CQRSAIFunction(Type cqrsObjectType) : AIFunction
+{
+    public enum JsonSchemaSource
```

Comment:
This will be helluva hard to understand. Let's pick one and just go with it. If it ends up not working correctly, we will switch centrally here. There is no point in maintaining both if we don't have a sensible use-case.

Reply:
Fair, I'll just remove it.